packages/ui/src/components/ResourceBlock/ResourceBlock.tsx

```diff
@@ -108,19 +108,27 @@ const Root = ({
 // an icon pinned to the top-right to show the completion state
 const StateIcon = ({ state }: { state: ItemProps["state"] }) => {
   const commonStyles: HydrogenAttributes = {
-    "data-h2-width": "base(20px)",
-    "data-h2-height": "base(20px)",
+    "data-h2-width": "base(x0.75)",
+    "data-h2-height": "base(x0.75)",
     "data-h2-position": "base(absolute)",
     "data-h2-location": "base(x0.75, x0.75, auto, auto)",
```

Comment:
The status icon isn't lining up with the title, also since the status icon is absolute positioned the text can still overlap with icon with long text (even if you added extra padding). The [absolute positioned](https://developer.mozilla.org/en-US/docs/Web/CSS/position#absolute) icon is pulled out of the normal document flow, and will end up just stacking on top of the text.
![image](https://github.com/user-attachments/assets/9b15e00c-8f6b-4a6a-b6be-25d2c2f8e36e)

I would recommend moving away from absolute positioning in this case (so that the text and status never overlap), and move the location of the status icon after the link and use flexbox to line then up, followed by the description text. I tried this below.

```tsx
    <div
      data-h2-display="base(flex)"
      data-h2-background="base(foreground)"
      data-h2-padding="base(x1) l-tablet(x1 x1.5)"
      data-h2-flex-direction="base(column)"
      data-h2-gap="base(x0.15)"
      data-h2-border-bottom="base:all:selectors[:not(:last-child)](1px solid gray.lighter)"
      data-h2-border-radius="base:all:selectors[:last-child](0 0 rounded rounded)"
      // make the containing block for state icon
      data-h2-position="base(relative)"
      {...extraStateStyles}
    >
      // ADD WRAPPING DIV AROUND LINK AND STATUS - Yoni
      <div
        data-h2-display="base(flex)"
        data-h2-align-items="base(flex-start)"
        data-h2-justify-content="base(space-between)"
      >
        <Link
          href={href}
          color="black"
          // yuck, style exception 😞
          data-h2-font-weight="base(bold)"
          // big click target black magic 🧙
          data-h2-position="base:selectors[::after](absolute)"
          data-h2-content="base:selectors[::after](' ')"
          data-h2-inset="base:selectors[::after](0)"
          data-h2-justify-self="base(end)"
          // icon extends margin + icons size: x0.75 + x0.75 = x1.5
          // item margin is base(x1) l-tablet(x1.5)
          data-h2-padding-right="base(x0.5) l-tablet(0)"
        >
          {title}
          {/* issue #11284 */}
          <ArrowSmallRightIcon // eslint-disable-line deprecation/deprecation
            data-h2-width="base(auto)"
            data-h2-height="base(x0.75)"
            data-h2-margin="base(x0.15)"
            data-h2-vertical-align="base(top)"
            aria-hidden
          />
        </Link>
        <StateIcon state={state} />
      </div> // END OF WRAPPING DIV
      <p data-h2-color="base(black.light)" data-h2-font-size="base(caption)">
        {description}
      </p>
    </div>
```
Then you can remove the absolute positioning in the Status component above.

```tsx
const commonStyles: HydrogenAttributes = {
    // Adding the min and max height was only way I found to keep the icon steady
    "data-h2-min-width": "base(x.75)",
    "data-h2-min-height": "base(x.75)",
    "data-h2-max-width": "base(x.75)",
    "data-h2-max-height": "base(x.75)",
  };

  if (state === "incomplete") {
    return (
      <ExclamationCircleIcon
        data-h2-color="base(error) base:dark(error.lighter)"
        {...commonStyles}
      />
    );
  }
  if (state === "complete") {
    return (
      <CheckCircleIcon
        data-h2-color="base(success) base:dark(success.lighter)"
        {...commonStyles}
      />
    );
  }
  return null;
```

Reply:
I don't think the status icon is supposed to line up with the text.  If you check Figma you can see that they don't line up and the icon is positioned from the top-right corner of the card.
![image](https://github.com/user-attachments/assets/a8103de8-bb57-4138-a99c-b92049e3a78b)

I added some right padding to the text line to ensure it wraps before overlapping with the icon.
![image](https://github.com/user-attachments/assets/1836184e-f4e5-4fdb-96b9-b31a17ce452e)

Reply:
Oh wow my bad :+1: It is pushed up into the center in the design.. however the padding isn't going to effect the icon since it's absolute positioned and is basically stacked on top of the other content in the div and would overlap it. I did test it thoroughly and it never overlapped the text, so I'm fine leaving it like this.